### PR TITLE
[CI] Setup buildkite

### PR DESCRIPTION
**Pull request by amontoison**
We could test `MadNLPGPU.jl` in a different PR.

### CODE REVIEW

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+steps:
+  - label: "Julia v1.10"
+    plugins:
+      - JuliaCI/julia#v1:
+          version: "1.10"
+      - JuliaCI/julia-test#v1: ~
+    agents:
+      queue: "juliagpu"
+      cuda: "*"
```
Comment (sshin23) [inline .buildkite/pipeline.yml:9]: we don't need nvidia GPUs for MadNLPTesting (only needed for MadNLPGPU). Wouldn't this be waste of resources?

Comment (amontoison) [reply .buildkite/pipeline.yml:9]: It's a waste of ressources but I don't in which subpackages you are using `GPU`.

Comment (sshin23) [reply .buildkite/pipeline.yml:9]: We only need it for testing `MadNLPGPU`

File: .buildkite/pipeline.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+steps:
+  - label: "Julia v1.10"
+    plugins:
+      - JuliaCI/julia#v1:
+          version: "1.10"
+      - JuliaCI/julia-test#v1: ~
```
Comment (amontoison) [inline .buildkite/pipeline.yml:6]:
```suggestion
      - JuliaCI/julia-test#v1: ~/lib/MadNLPGPU
```

File: .buildkite/pipeline.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
+steps:
+  - label: "Julia v1.10"
+    plugins:
+      - JuliaCI/julia#v1:
+          version: "1.10"
+      - JuliaCI/julia-test#v1: ~/lib/MadNLPGPU
```
Comment (amontoison) [inline .buildkite/pipeline.yml:6]:
```suggestion
      - JuliaCI/julia-test#v1: ~
```

File: .buildkite/pipeline.yml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    agents:
+      queue: "juliagpu"
+      cuda: "*"
```
Comment (amontoison) [inline .buildkite/pipeline.yml:9]:
```suggestion
      cuda: "*"
    working_directory: "~/lib/MadNLPGPU"
```

**Comment by sshin23** [comment]
Right now, I don't think we'll need buildkite as the current CI is working

**Pull request closed without merging**